client/my-sites/current-site/domain-to-paid-plan-notice.jsx

```diff
@@ -0,0 +1,49 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+import {
+	get,
+	noop,
+} from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import abtest from 'lib/abtest';
+import { eligibleForDomainToPaidPlanUpsell } from 'state/selectors';
+
+export class DomainToPaidPlanNotice extends Component {
+	static propTypes = {
+		translate: PropTypes.func.isRequired,
+	}
+
+	static defaultProps = {
+		translate: noop,
+	}
+
+	render() {
+		const { eligible, translate } = this.props;
```

Comment:
Does this component meet all of its data needs from the current site being populated? Or do we need to add a QueryComponent here for `eligibleForDomainToPaidPlanUpsell` to work properly on a clean cache? eg does it show up properly with `localStorage.clear(); indexedDB.deleteDatabase( 'calypso' );`